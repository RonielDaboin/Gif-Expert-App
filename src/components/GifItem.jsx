import React from 'react'
import PropTypes from 'prop-types'

const GifItem = ({id, title, url}) => {
  return (
    <div className='card'> 
      <img src={ url } alt={ title } aria-placeholder={ id }/>
      <p> { title } </p>
    </div>
  )
}

export default GifItem

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}