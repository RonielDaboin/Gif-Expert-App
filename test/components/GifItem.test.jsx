import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"

describe('Pruebas en <GifItem/>', () => {

    const title = 'nvdia news'

    const url = 'https://nvidia-news.com/nvidia-news'
  
    test('Esto deberia hacer match con el snapshot ', () => {
      
        const { container } = render( <GifItem title={title} url={url}/> );
        
        expect ( container ).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
      
        render( <GifItem title={title} url={url}/> );
        //screen.debug();

        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

    })

    test('Esto deberia mostrar el titulo', () => {
       
        render( <GifItem title={title} url={url}/> );
        expect( screen.getByAltText( title )).toBeTruthy();

    })
    
    

})
