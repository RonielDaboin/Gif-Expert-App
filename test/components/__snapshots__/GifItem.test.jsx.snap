// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Esto deberia hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="nvdia news"
      src="https://nvidia-news.com/nvidia-news"
    />
    <p>
       
      nvdia news
       
    </p>
  </div>
</div>
`;
